PHP: ensure 5.3 compatibility

virtualtam opened the pull request:
Relates to #250

Modifications
- add PHP 5.3 to Travis environments
- rewrite array declarations: explicitely use array() instead of []
- move checkPHPVersion to application/Utils.php
- move timezone functions to application/TimeZone.php
- improve test coverage
- bump required version from 5.1.0 to 5.3.x

ArthurHoaro commented:
You forgot L949. But other than that everything seems to be working with PHP 5.3.3.

virtualtam commented:
Actually, I think a couple methods can also be simplified and covered by tests:
* `isTZvalid()`
* `templateTZform()`

Some comments are outdated, too, there are references to 5.1.x compatibility

Working on it! :)

virtualtam commented:
@ArthurHoaro I've done some additional edits and cleanup, could you confirm it works with a 5.3.x version?

ArthurHoaro commented on application/TimeZone.php:
```diff
@@ -0,0 +1,91 @@
+        $continents_html.='<option  value="'.$continent.'"'.($pcontinent==$continent?' selected':'').'>'.$continent.'</option>';
+    }
+
+    $cities_html = $cities[$pcontinent];
```
`$citites_html` is never used.

virtualtam replied:
kenavo, $cities_html :)

ArthurHoaro commented on application/TimeZone.php:
```diff
@@ -0,0 +1,91 @@
+ *  list($htmlform, $js) = templateTZform('Europe/Paris');
+ *
+ * @param $preselected_timezone  preselected timezone (can be 'UTC/UTC') (optional)
+ * @return array(html, js)
```
A quick description of what it contains would be better.

virtualtam replied:
whole docstring rewritten, code comments added

ArthurHoaro commented on tests/TimeZoneTest.php:
```diff
@@ -0,0 +1,49 @@
+     */
+    public function testGenerateTimeZoneForm()
+    {
+        generateTimeZoneForm();
```
Since `generateTimeZoneForm` isn't suppose to raise Exception, you should check return value.

virtualtam replied:
checks added for generated content, both HTML and Javascript

ArthurHoaro commented:
Aside from my minor comments, everything seems to be working with PHP 5.3.3 and unit tests are OK.

virtualtam commented:
Thanks for reviewing!
I'll see how the TZ methods can be improved (e.g. not only coding style) and write a bit more clever tests :)

ArthurHoaro commented:
Perfect. Also, remind to update README.

virtualtam commented:
Comments taken into account, here are the more significant changes since the last proposed PR:
- update README
- refactor TZ code for readability, more relevant docstrings and inline comments
- more in-depth unit tests: check generated HTML and Javascript content

ArthurHoaro commented on application/TimeZone.php:
```diff
@@ -0,0 +1,111 @@
+    }
+
+    $tz = $continent.'/'.$city;
+    if (in_array($tz, timezone_identifiers_list())) {
```
That's a bit picky, but you can just `return in_array`.

virtualtam replied:
thanks, I'll re-submit :)

ArthurHoaro commented:
Great job! I'll make a final test in 5.3 and we can merge this. :)

virtualtam commented:
Thanks for testing :)
